nomad/structs/structs.go

```diff
@@ -5386,6 +5387,26 @@ type NamespaceCapabilities struct {
 	DisabledTaskDrivers []string
 }
 
+// NamespaceNodePoolConfiguration stores configuration about node pools for a
+// namespace.
+type NamespaceNodePoolConfiguration struct {
+	// Default is the node pool used by jobs in this namespace that don't
+	// specify a node pool of their own.
+	Default string
+
+	// Allowed is a list of glob patterns to match node pool names that jobs in
+	// this namespace are allowed to use. If defined, node pools that do not
+	// match any of these patterns are not allowed to be used.
+	// Conflicts with Denied.
```

Comment:
Minor nitpick: we say "conflicts with denied" here which is a little more vague than the "cannot be used with" that the docs say. It might be nice in these internal docstrings to be clear that they can't be set at the same time (or they get merged, or whatever the behavior ultimately becomes once the ENT work ships).

Reply:
Yeah, I think I wrote these first then wrote a better version in the docs. I will make them the same.